index.js

```diff
@@ -76,10 +56,11 @@ function compatibleConstructor(thrown, errorLike) {
  */
 
 function compatibleMessage(thrown, errMatcher) {
-  var comparisonString = typeof thrown === 'string' ? thrown : thrown.message;
-  if (errMatcher instanceof RegExp) {
+  var errMatcherType = type(errMatcher).toLowerCase();
```

Comment:
If we are literally just comparing `regexp` and `string`, a `typeof`/`instanceof` check would be better (faster) here.